DO-NOT-MERGE Add implementation of ipython widget

chmp opened the pull request:
As discussed in https://github.com/vega/ipyvega/issues/93, this PR contains an implementation for an IPython Vega widget. In addition to the existing display system, it allows

- to put altair charts in a larger UI with other widgets,
- to change the chart without re-displaying the widget, and
- to dynamically update the data via changesets.

I struggled a bit with webpack to get it to only optionally define the widget component. Not sure whether my solution is idiomatic.

domoritz commented on src/index.ts:
```diff
@@ -92,3 +92,56 @@ export function render(
     .catch(error => showError(el, error));
 }
+
+// optionally define a Jupyter Widget if the plugin is available available
```
double available

domoritz commented on src/index.ts:
```diff
@@ -92,3 +92,56 @@ export function render(
+    view: any
+
+    render() {
+          const reembed = () => {
```
please use 2 spaces

domoritz commented on vega/widget.py:
```diff
@@ -0,0 +1,72 @@
+
+    Usage with ``altair``::
+
+        widget = VegaWidget.from_cart(chart)
```
from cart, lol ;-)

domoritz commented on src/index.ts:
```diff
@@ -92,3 +92,56 @@ export function render(
+              const filter = new Function('datum', 'return (' + (ev.remove || 'false') + ')');
+              const newValues = ev.insert || [];
+
+              const changeSet = (
```
why not just 

```ts
const changeSet = this.view.changeset().insert(newValues).remove(filter)
```

You can access `vegaEmbed.vega.changeset()` as well but this is fine.

domoritz commented on src/index.ts:
```diff
@@ -92,3 +92,56 @@ export function render(
+export const VegaWidget = import('@jupyter-widgets/base')
+.then(widgets => {
+  return class VegaWidget extends widgets.DOMWidgetView {
+    view: any
```
make this `View` from Vega

chmp commented:
@domoritz my yarn upgraded me to `vega-embed==3.16.0` and now `Mode` is no longer available for import. Could you give a pointer how to update the import in the first line of `index.ts`? Thanks. 

Once I get it to compile again, I will push commits addressing your points above :).

domoritz commented:
Ahh, let me fix that in Embed.

domoritz commented:
Sorry about that. Please update to 3.16.1.

domoritz commented:
Can you rebase against master? this will update all the dependencies.

chmp commented:
Hm. I have to backtrack here. I used a new venv to check everything and it seems that the optional dependency part within webpack is not working as expected in production mode. Now, it seems extensions are loaded in a different order and the whole vega extension is no longer loaded.
